server/text_generation/utils/tokens.py

```diff
@@ -47,6 +50,11 @@ def __init__(
         # the following idea is largely copied from this PR: https://github.com/huggingface/transformers/pull/5420/files
         # all samplers can be found in `generation_utils_samplers.py`
         sampling = do_sample
+
+        if watermark:
+            warpers.append(WatermarkLogitsProcessor(vocab_size, device=device))
```

Comment:
I don't see kwargs (like `delta`, `gamma`) being passed at construction time here. Does this mean that a fixed version of the watermark will be instantiated for all generations?

Reply:
Yes indeed.